extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -827,6 +831,171 @@ export class ProjectsController {
+				defaultUri: vscode.workspace.workspaceFolders?.[0]?.uri,
+				title: constants.selectProjectLocation
+			});
+
+			if (!folders) {
+				return;
+			}
+
+			outputFolder = folders[0].fsPath;
+			projectName = path.basename(specPath, constants.yamlFileExtension);
+			newProjectFolder = path.join(outputFolder, projectName);
+
+			if (await utils.exists(newProjectFolder)) {
+
+				quickpickSelection = await vscode.window.showQuickPick(
+					[constants.browseEllipsis],
+					{ title: constants.folderAlreadyExistsChooseNewLocation(newProjectFolder), ignoreFocusOut: true });
+				if (!quickpickSelection) {
+					return;
+				}
+			} else {
+				valid = true;
+			}
+		}
+
+		await fs.mkdir(newProjectFolder);
+		return { newProjectFolder, outputFolder, projectName };
+	}
+
+	public async generateAutorestFiles(specPath: string, newProjectFolder: string): Promise<void> {
+		await this.autorestHelper.generateAutorestFiles(specPath, newProjectFolder);
+	}
+
+	public async openProjectInWorkspace(projectFilePath: string): Promise<void> {
+		const workspaceApi = utils.getDataWorkspaceExtensionApi();
+		await workspaceApi.validateWorkspace();
+		await workspaceApi.addProjectsToWorkspace([vscode.Uri.file(projectFilePath)]);
+
+		workspaceApi.showProjectsView();
+	}
+
+	public async generateProjectFromOpenApiSpec(): Promise<Project | undefined> {
+		try {
+			// 1. select spec file
+			const specPath: string | undefined = await this.selectAutorestSpecFile();
+			if (!specPath) {
+				return;
+			}
+
+			// 2. select location, make new folder
+			const projectInfo = await this.selectAutorestProjectLocation(specPath!);
+			if (!projectInfo) {
+				return;
+			}
+
+			// 3. run AutoRest to generate .sql files
+			await this.generateAutorestFiles(specPath, projectInfo.newProjectFolder);
+
+			// 4. create new SQL project
+			const newProjFilePath = await this.createNewProject({
+				newProjName: projectInfo.projectName,
+				folderUri: vscode.Uri.file(projectInfo.outputFolder),
+				projectTypeId: constants.emptySqlDatabaseProjectTypeId
+			});
+
+			const project = await Project.openProject(newProjFilePath);
+
+			// 5. add generated files to SQL project
+			let fileFolderList: vscode.Uri[] = await this.getSqlFileList(project.projectFolderPath);
+			await project.addToProject(fileFolderList.filter(f => !f.fsPath.endsWith(constants.autorestPostDeploymentScriptName))); // Add generated file structure to the project
+
+			const postDeploymentScript: vscode.Uri | undefined = this.findPostDeploymentScript(fileFolderList);
+
+			if (postDeploymentScript) {
+				await project.addScriptItem(path.relative(project.projectFolderPath, postDeploymentScript.fsPath), undefined, templates.postDeployScript);
+			}
+
+			// 6. add project to workspace and open
+			await this.openProjectInWorkspace(newProjFilePath);
+
+			return project;
+		} catch (err) {
+			void vscode.window.showErrorMessage(`${constants.generatingProjectFailed}: ${utils.getErrorMessage(err)}`);
```

Comment:
You shouldn't be combining strings like this since not all languages are LTR. You should make a separate loc constant that takes in the error message - e.g. https://github.com/microsoft/azuredatastudio/blob/main/extensions/arc/src/localizedConstants.ts#L259